### PR TITLE
fix(content-group): adding h-tags to headings

### PR DESCRIPTION
**Pull request by marcelojcs**
### Related Ticket(s)

[Jira](https://jsw.ibm.com/browse/ADCMS-4630)

>The H3 and H4 headings are not properly configured in the AEM template. They should be tagged as the H1 and H2 (from the same AEM template). Otherwise, they won't have the same SEO relevancy because it's not recognized as H2 & H3.
### Description

Adding the respective H tags to content-group-heading and content-item-heading components.

### CODE REVIEW

**Comment by marcelojcs** [comment]
# V2 version of this PR
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11739

File: packages/web-components/src/components/content-group/content-group-heading.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ class DDSContentGroupHeading extends StableSelectorMixin(LitElement) {
   }
 
   render() {
-    return html` <slot></slot> `;
+    return html` <h3><slot></slot></h3> packages/web-components/src/components/content-group/content-group-heading.ts`;
```
Comment (andy-blum) [inline packages/web-components/src/components/content-group/content-group-heading.ts:42]:
```suggestion
    return html` <h3><slot></slot></h3>`;
```

**Review by andy-blum — CHANGES_REQUESTED** [review]

**Comment by andy-blum** [comment]
It seems that in addition to adding the h3/h4 elements to the shadowroot, we'll need to remove the `role` and `aria-level` properties in use here. As it is now, the headings appear twice in screen readers.

Screenshot 2024-04-26 at 2 29 13 PM

**Review by jkaeser — APPROVED** [review]
Code changes here look good to me!

**Comment by andy-blum** [comment]
There are some subtle but noticable differences in the typography with the new heading elements. Can we get some designer input on what this is supposed to be rendering as?

Screenshot 2024-05-06 at 4 26 57 PM
Screenshot 2024-05-06 at 4 27 37 PM

**Comment by RichKummer** [comment]
Hey @marcelojcs , I'm seeing the heading for `content item` rendering differently than it should. It should use the fixed`$heading-02` token for the heading in the v1 package.

**This PR:**
Screenshot 2024-05-06 at 4 27 39 PM

**Spec samples:**
Screenshot 2024-05-06 at 4 34 10 PM

**Comment by RichKummer** [comment]
Hey @marcelojcs and @andy-blum , 

`content item` looks fixed. I noticed that for `content group`, the heading does not appear to be updating its size as `$expressive-heading-04` across breakpoints. It should be `28px` up to `@lg` and then update to `32px` for `@lg – @max`.

Right now, I only see the heading render as `28px` across breakpoints.

**Comment by marcelojcs** [comment]
@RichKummer 
Addressed in my latest commit.

**Comment by RichKummer** [comment]
Thanks @marcelojcs ! Both `content item` and `group` look fixed. I did see that for some reason the `content item horizontal`'s header updated to a different type token than it should.

It should appear with the heading as `$expressive-heading-03` 

Screenshot 2024-05-08 at 1 29 12 PM

**Comment by andy-blum** [comment]
@RichKummer updated font rules to inherit their `font` `letter-spacing` and `line-height` rules from the parent components.

**Review by RichKummer — APPROVED** [review]
@marcelojcs @andy-blum Thanks for the fixes! LGTM

**Review by andy-blum — APPROVED** [review]

**Pull request merged**